src/coreclr/debug/daccess/dacdbiimpl.cpp

```diff
@@ -5650,10 +5650,13 @@ void DacDbiInterfaceImpl::LookupEnCVersions(Module*          pModule,
     DebuggerJitInfo * pDJI = NULL;
     EX_TRY_ALLOW_DATATARGET_MISSING_MEMORY
     {
-        pDMI = g_pDebugger->GetOrCreateMethodInfo(pModule, mdMethod);
-        if (pDMI != NULL)
+        if (g_pDebugger != NULL)
```

Comment:
How do we get to this state? This gets allocated in EEStartup. Is the DBI getting called super early on? Feels like this is something that shouldn't happen. Data target missing memory like the catch gets should be the main reason to not have it (or init error)

Reply:
This is a good overall cleanup for defensive coding.  To be clear for this PR, I did not encounter this with a real world example.  It would be fine with me to not backport this particular fix and just leave the fix shipping for v.next.  It's only the `collections.cpp` issue that I'd like to backport since I ran into it.

Reply:
@hoyosjs I hit those particular issues when debugging CLR startup scenarios. ... I spend a lot of time debugging the runtime startup path, and I regularly was hitting DAC crashes when trying to diagnose the startup path. Typically, the DAC isn't super useful at that point, but its really annoying to be unable to step through the various process creation paths.

Reply:
Also, yes, the DBI is commonly called this early in the newer Windbg UI.

Reply:
I withdraw my comment.  :)  If this happens in CLR startup I do recommend backporting this part to .net 8 as well.  We deal with that often in servicing, watson and elsewhere.  Even if most CLR devs will be on .Net 9, it would be good to get ahead of this issue for our long term servicing release.